addons/mail/static/src/discuss/gif_picker/common/gif_picker.js

```diff
@@ -166,24 +166,31 @@ export class GifPicker extends Component {
             return;
         }
         this.state.loadingGif = true;
+        const params = {
+            country: this.userService.lang.slice(3, 5),
+            locale: this.userService.lang,
+            search_term: this.state.searchTerm,
+        };
+        if (this.next) {
+            params.position = this.next;
+        }
         try {
-            const params = {
-                country: this.userService.lang.slice(3, 5),
-                locale: this.userService.lang,
-                search_term: this.state.searchTerm,
-            };
-            if (this.next) {
-                params.position = this.next;
+            const res = await this.sequential(() =>
+                this.rpc("/discuss/gif/search", params, {
+                    silent: true,
+                })
+            );
+            if (!res) {
+                return;
             }
-            const { results, next } = await this.rpc("/discuss/gif/search", params, {
-                silent: true,
-            });
-            if (results) {
+            const { next, results } = res;
+            if (results && next) {
```

Comment:
I'm not sure what it is but it looks wrong to also ensure `next` to push current results?

Reply:
It's not really useful, it will always exist with results.